policy-service/src/main/java/uk/gov/gchq/palisade/service/policy/stream/SerDesConfig.java

```diff
@@ -0,0 +1,146 @@
+ */
+
+package uk.gov.gchq.palisade.service.policy.stream;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.common.serialization.Serializer;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.springframework.core.serializer.support.SerializationFailedException;
+
+import uk.gov.gchq.palisade.service.policy.model.AuditErrorMessage;
+import uk.gov.gchq.palisade.service.policy.model.PolicyRequest;
+import uk.gov.gchq.palisade.service.policy.model.PolicyResponse;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+
+/**
+ * Static configuration for kafka key/value serialisers/deserialisers
+ * - Each input has a pair of key/value deserialisers
+ * - Each output has a pair of key/value serialisers
+ * In general, the keys are not used so the choice of serialiser is not important
+ */
+public final class SerDesConfig {
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+    private static final String SERIALIZATION_FAILED_MESSAGE = "Failed to serialize ";
+    private static final String DESERIALIZATION_FAILED_MESSAGE = "Failed to deserialize ";
+
+    private SerDesConfig() {
+        // Static collection of objects, class should never be instantiated
+    }
+
+    /**
+     * Kafka key serialiser for upstream messages coming in as input
+     * Used by the Rest Controller to insert requests onto the topic
+     *
+     * @return an appropriate key serialiser for the topic's message content
+     */
+    public static Serializer<String> resourceKeySerializer() {
+        return new StringSerializer();
+    }
+
+    /**
+     * Kafka value serialiser for upstream messages coming in as input
+     * Used by the Rest Controller to insert requests onto the topic
+     *
+     * @return an appropriate value serialiser for the topic's message content (AttributeMaskingRequest)
+     */
+    public static Serializer<PolicyRequest> resourceValueSerializer() {
+        return (String ignored, PolicyRequest policyRequest) -> {
+            try {
+                return MAPPER.writeValueAsBytes(policyRequest);
+            } catch (IOException e) {
+                throw new SerializationFailedException(SERIALIZATION_FAILED_MESSAGE + policyRequest.toString(), e);
+            }
+        };
+    }
+
+    /**
+     * Kafka key deserialiser for upstream messages coming in as input
+     *
+     * @return an appropriate key deserialiser for the topic's message content
+     */
+    public static Deserializer<String> resourceKeyDeserializer() {
+        return new StringDeserializer();
+    }
+
+    /**
+     * Kafka value deserialiser for upstream messages coming in as input
+     *
+     * @return an appropriate value deserialiser for the topic's message content (AttributeMaskingRequest)
```

Comment:
AttributeMaskingRequest in JavaDoc should be PolicyRequest

Reply:
removed